### ISSUE TITLE
game page crashes frequently

### ISSUE THREAD

**Issue by edwardbrook** [KC3Kai/KC3Kai#2638]
68.0.3440.106 (Official Build) (64-bit)
Windows 10 Pro
KanColle Command Center 改 33.1.1
Customize Mode

![capture1 - copy](https://user-images.githubusercontent.com/6615983/44590891-8f735700-a7ff-11e8-95cd-b534707f80ba.PNG)

happens 3 - 4 times per day
I'm sure that I have disabled recording network log every time I start game

**Comment by sinsinpub** [comment]
not sure if it is caused by memory issue, but sure not directly relevant to kc3.
try to disable chrome's auto tab discarding?
`chrome://flags/#automatic-tab-discarding`

**Comment by Dahzz** [comment]
I'm having the same issue, never crashed before the latest update.

But then again, i just started using the KC3 desktop shortcut version.

**Comment by edwardbrook** [comment]
![capture3](https://user-images.githubusercontent.com/6615983/44617011-98673580-a89e-11e8-9338-956f993e68e5.PNG)
@sinsinpub 
already disabled chrome's auto tab discarding
still crashed

**Comment by sinsinpub** [comment]
information not enough tho, from ur screenshot, known fact is: 'crashed' thing is the kc game iframe page, both dmm page and kc3 page not 'crashed'.

since game memory leak is known for some environment, u might try this: turn all in-game sound volume off at home port screw button (drag 3 volume bars to leftmost)

or if u wanna prove crash actually caused by kc3: disable kc3 in chrome extension management page to see if it never crash again

**Comment by edwardbrook** [comment]
it seems that turning off in-game sound is a solution, 2 days gaming without any crash, THX bro

**Comment by sinsinpub** [comment]
@astcd2 i don't think u were using windows task manager correctly. at least u should monitor the exact process which is assigned to game sub frame by showing the `Process ID` (or `PID`).

for short, the 'Memory footprint' taken by chrome's process (by site) is not shrink-able by design. so as long as the page (or sub frame) not closed or reloaded, the memory value u see will never be reduced.

u'd better monitor it via chrome's task manager (shortcut is `Shift + Esc`)
this is mine (the same chrome version) after game is just newly started:
![image](https://user-images.githubusercontent.com/160240/45803054-1e826a80-bceb-11e8-944f-d1a4c6b26441.png)
the `Subframe: http://<your game server ip>` the is actually sub frame process where game is running in.
and the `Tab: DevTools ...` is the process of devtools we opened for KC3

with the in-game bgm volume enabled, after 10 more bgm switches (by switching between fleet compos and resupply repeatedly), the game sub frame process memory footprint grown quickly. the same value can be monitored by windows task manager.
![image](https://user-images.githubusercontent.com/160240/45803264-bc763500-bceb-11e8-8a36-4db25b8fbd8c.png)

after closing KC3 panel (the devtools), `Tab: Devtools` process disappeared, and game sub frame not down much.
![image](https://user-images.githubusercontent.com/160240/45803429-2989ca80-bcec-11e8-8ae2-53960e358842.png)

keep devtools panel closed, continue to switch bgm more times, game sub frame process can grow still. after long game play time enough, memory footprint can reach the 'crash' threshold.
![image](https://user-images.githubusercontent.com/160240/45803706-d6fcde00-bcec-11e8-8336-9e4de42f5a8e.png)

then after reloading the whole game page, eyes on the Process ID of the sub frame and DMM tab: DMM page PID not changed but sub frame is in new process now, ofc memory footprint is brand new value.
![image](https://user-images.githubusercontent.com/160240/45803894-4d014500-bced-11e8-9ba9-4a6d4ba4db3c.png)

**Comment by astcd2** [comment]
@sinsinpub Thanks for the quick response and your work on KC3! I have tried the above test and got different results.

1. Here is the Chrome task manager after a fresh start with KC3 open (in game sound is on). The game subframe is using about 160MB:

![untitled1](https://user-images.githubusercontent.com/43430766/45807099-09670500-bd06-11e8-9731-33ee2dbd2df7.png)

2. Here it is after about 20 minutes with KC3 open. The game subframe is using about 673MB now.

![untitled2](https://user-images.githubusercontent.com/43430766/45807268-7a0e2180-bd06-11e8-8de4-15f7830aa62c.png)

3. Here it is about 30 seconds after I close the KC3 panel. The game subframe has dropped back to 291MB (no change to process ID).

![untitled3](https://user-images.githubusercontent.com/43430766/45807372-b04ba100-bd06-11e8-8a14-512ac0c77ad2.png)

4. Here it is after another 15 minutes or so playing the game with the KC3 panel closed. The memory usage for the game subframe is remaining generally constant.

![untitled4](https://user-images.githubusercontent.com/43430766/45807823-a8d8c780-bd07-11e8-8bdc-a8bc6de1fdda.png)

For the purposes of this test, all I did was repeatedly sortie to 5-2-1, refuel and send out enseis.

As this issue doesn't seem to affect all users, I wonder if there is another factor in my operating environment which could be causing it?

Cheers

**Comment by sinsinpub** [comment]
@astcd2 yep, i'm definitely interest in ur result. not all environments suffering is known tho.
could u paste ur `chrome://version` here and list the other extensions u enabled?

**Comment by astcd2** [comment]
@sinsinpub 

Here is the chrome://version:

Google Chrome | 69.0.3497.100 (Official Build) (64-bit) (cohort: Stable)
-- | --
Revision | 8920e690dd011895672947112477d10d5c8afb09-refs/branch-heads/3497@{#948}
OS | Windows
JavaScript | V8 6.9.427.23
Flash | 31.0.0.108 C:\Users\TC\AppData\Local\Google\Chrome\User Data\PepperFlash\31.0.0.108\pepflashplayer.dll
User Agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36
Command Line | "C:\Program Files (x86)\Google\Chrome\Application\chrome.exe" --profile-directory="Profile 1" --flag-switches-begin --disable-quic --enable-features=AutomaticTabDiscarding --flag-switches-end
Executable Path | C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
Profile Path | C:\Users\TC\AppData\Local\Google\Chrome\User Data\Profile 1
Variations | c134752e-1ece35533e006338-3f4a17df1a0d11d4-2f9febdfe202a358-ca7d8d80ebeb14fc-3f4a17df752a9400-3d47f4f4b7e2524c-ca7d8d808fe39baa-3f4a17df3095aa95-3f4a17dfc27fec31-2d5b6ed97c1bc906-f55a797447e5d3db-3d47f4f49ca1387e-ca7d8d801149accc-f23d1dea4dc30737-b8a5ea08c865fdc1-ca7d8d8015d89564-80f9a33ea582a1b8-ad75ce178ee5ed19-ca7d8d8074658432-ca7d8d803042ad4b-ca7d8d80ebbb4e0a-ca7d8d80e56c5101-ad2fa222267255c3-f4950e99249dd49a-20131bcc44827ee5-f23d1dea88a387d2-ee748cef8f1e27f-ca7d8d80edbcf7c5-1cc1312cde47491b-33c3eba543f62d3b-28165b593a0563a1-65222f0b9e5c75f1-e8719b9a6872f671-991e1e1332bb593-3d47f4f4f79cb77b-3f4a17df4ea303a6-2e46ed916e6e0c7e-3f17a7d8d92562a9-4d2fac877aa46da5-c946b1504da5ae82-91c810ef2c1d398c-3f4a17df6973a1cf-3f4a17dfcc54eb06-ca7d8d80cac0a91c-7766273758a025e3-36e97b2cdf072bba-ca7d8d80ff29b1bd-37ef7e174bc337ce-694658969a2f4e5b-ca7d8d80ddf77e2c-ca7d8d801354da85-f1a864dc17507c76-ca7d8d80494d8760-52325d43f47ae82a-746c2ad43ac60855-486e2a9cf296190c-4c0731544442aae2-4ad60575ed1d377-e1cc0f1475f0f0a0-d7f6b13ce2b18481-7158671ee7e71889-e1cc0f14b1ceb06f-d13723343a4029d-ca7d8d806ab94979-3f4a17df94e68624-803f8fc48834fcca-ca7d8d80ea0f933d-ca7d8d80
Linker | lld-link

Here are the Chrome extensions:

![untitled](https://user-images.githubusercontent.com/43430766/45812509-53a2b300-bd13-11e8-8f20-5f7754573624.png)
